File: 2024/august/javaScript/04.js
/**
 * @param {number[]} nums
 * @param {number} n
 * @param {number} left
 * @param {number} right
 * @return {number}
 */
var rangeSum = function(nums, n, left, right) {
        const allSums = Array.from({length : Math.floor(n * (n+1) / 2)} , ()=> 9);
        let k = 0;
        for (let i = 0; i < n; ++i){
            let curSum = 0;
            for (let j = i; j < n; ++j){
                curSum+= nums[j];
                allSums[k++] = curSum;
            }
        }
        allSums.sort((a , b)=> a-b);
        let ans = 0;
        const mod = 1e9+7;
        for (let i = --left; i < right ; ++i) {
            ans = (ans+ allSums[i]) % mod;
        }
        return ans;
};

File: 2024/august/java/04.java
class Solution {
    public int rangeSum(int[] nums, int n, int left, int right) {
        var allSums = new int[n * (n+1) / 2];
        int k = 0;
        for (int i = 0; i < n; ++i){
            int curSum = 0;
            for (int j = i; j < n; ++j){
                curSum+= nums[j];
                allSums[k++] = curSum;
            }
        }
        Arrays.sort(allSums);
        int ans = 0;
        assert(k == n * (n+1) / 2);
        final int mod = (int)1e9+7;
        for (int i = --left; i < right ; ++i) {
            ans = (int)((long)ans+ allSums[i]) % mod;
        }
        return ans;
        
    }
}
